featuretools/synthesis/dfs.py

```diff
@@ -265,3 +301,48 @@ def dfs(entities=None,
                                               progress_callback=progress_callback,
                                               include_cutoff_time=include_cutoff_time)
     return feature_matrix, features
+
+
+def get_unused_primitives(specified, features):
+    """Get a list of unused primitives based on a list of specified primitives and a list of output features"""
+    if not specified:
+        return []
+    specified = {primitive if isinstance(primitive, str) else primitive.name for primitive in specified}
+    used = {feature.primitive.name for feature in features}
+    return sorted(list(specified.difference(used)))
+
+
+def warn_unused_primitives(unused_primitives):
+    trans_unused, agg_unused, groupby_unused, where_unused = unused_primitives
+    unused_string = ""
+    if trans_unused:
+        unused_string += " trans_primitives: {}".format(trans_unused)
+    if agg_unused:
+        unused_string += " agg_primitives: {}".format(agg_unused)
+    if groupby_unused:
+        unused_string += " groupby_trans_primitives: {}".format(groupby_unused)
+    if where_unused:
+        unused_string += " where_primitives: {}".format(where_unused)
+
+    warning_msg = "Some specified primitives were not used during DFS.{}. ".format(unused_string) + \
```

Comment:
For readability, would it be better to put each unused group indented on their own line? Something like:
> Some specified primitives were not used during DFS:
> &ensp;&ensp;trans_primitives: ['greater_than_scalar']
> &ensp;&ensp;agg_primitives: ['mode', 'num_unique']
> This may be caused by a using a value of max_depth that is too small, not setting interesting values, or it may indicate no compatible variable types for the primitive were found in the data.

Reply:
Yes, I think this would be better. I'll make this update.